src/vacuum_map_parser_ijai/beautify_min.py

```diff
@@ -0,0 +1,1158 @@
+# STRAIGHT REWRITE FROM MI HOME PLUGIN`S CODE
+import base64
+
+class Point(object):
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def __repr__(self) -> str:
+        return f'Point({self.x},{self.y})'
+
+
+class BeautifyMap:
+    def __init__(self):
+        self.map = []
+        self.tRect = {
+            "x": 0,
+            "y": 0,
+            "width": 0,
+            "height": 0
+        }
+        self.x_min = 0
+        self.x_max = 0
+        self.y_min = 0
+        self.y_max = 0
+        self.resolution = 0
+        self.size_x = 0
+        self.size_y = 0
+
+    def setSize(self, x_min1, x_max1, y_min1, y_max1, resolution1):
+        self.map = []
+        self.size_x = 0
+        self.size_y = 0
+        self.x_min = int(resolution1 * round(x_min1 / resolution1))
+        self.y_min = int(resolution1 * round(y_min1 / resolution1))
+        self.x_max = int(resolution1 * round(x_max1 / resolution1))
+        self.y_max = int(resolution1 * round(y_max1 / resolution1))
+        self.size_x = int(round((x_max1 - x_min1) / resolution1))
+        self.size_y = int(round((y_max1 - y_min1) / resolution1))
+        self.resolution = resolution1
+
+    def setMap(self, tMapStruct):
+        self.setSize(tMapStruct.mapHead.minX, tMapStruct.mapHead.maxX,
+                     tMapStruct.mapHead.minY, tMapStruct.mapHead.maxY, tMapStruct.mapHead.resolution)
+
+        mapdata_arr = tMapStruct.mapData.mapData
+
+        tempArray = [0] * len(mapdata_arr)
+
+        for i in range(len(mapdata_arr)):
+            if (mapdata_arr[i] > 127):
+                tempArray[i] = -128
+            else:
+                tempArray[i] = mapdata_arr[i]
+
+        self.map = tempArray
+
+    def getMap(self):
+        return self.map
+
+    def transform(self):
+        black_boundary = []
+        non_boundary_noise = []
+        all_region = []
+        self.tRect = self.findRoiMap(self.tRect)
+        self.expandBlackRect(4, 4, self.map[0], self.tRect)
+        self.expandWhiteRect(4, 4, self.map[0], self.tRect)
+        self.refineBoundary(0, 10, self.tRect)
+        non_boundary_noise = self.eliminateNonBoundaryNoise(
+            non_boundary_noise, self.tRect, 127, -128, 0)
+        self.expandSingleConvexBoundary(50, -128, 4, 4, self.tRect)
+        black_boundary = self.removeIndependentRegion(
+            all_region, black_boundary, 3, self.tRect)
+        non_boundary_noise = self.fillNonBoundaryNoise2(
+            non_boundary_noise, self.tRect)
+        self.refineBoundary(0, 10, self.tRect)
+        self.map = self.fillBlackComponent(self.map, black_boundary, -128)
+
+    def findRoiMap(self, rect):
+        top_bound = self.size_x
+        bottom_bound = 0
+        left_bound = self.size_y
+        right_bound = 0
+        for idx in range(self.size_x):
+            for idy in range(self.size_y):
+                if (self.map[idy * self.size_x + idx] != 0):
```

Comment:
Consider optimizing this complex condition. As a first step all "else"s can be painlessly eliminated.